shell_turtlestein.py

```diff
@@ -105,6 +105,10 @@ class ShellPromptCommand(sublime_plugin.WindowCommand):
     def run(self):
         if not hasattr(self, 'cmd_history'):
             self.cmd_history = []
+        for pre_cmd_dict in settings().get('pre_run_commands', []):
+            pre_cmd = pre_cmd_dict.get('command')
+            pre_cmd_args = pre_cmd_dict.get('args')
+            self.window.run_command(pre_cmd, pre_cmd_args)
```

Comment:
Do you want this to when the prompt is opened? Would it make more sense to run it just before the command is executed (i.e. in the `on_done` callback)?